core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -320,14 +321,35 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         .filter { entry =>
           try {
             val prevFileSize = fileToAppInfo.get(entry.getPath()).map{_.fileSize}.getOrElse(0L)
+
+            def canAccess = {
```

Comment:
Return type? Also I'd make this a top-level method (with the entry and action as parameters), maybe even in `SparkHadoopUtil`... just to avoid the deeply-nested method declaration. That allows you to easily write a unit test for it (yay!).